plugin-server/src/utils/db/db.ts

```diff
@@ -642,10 +642,23 @@ export class DB {
         isIdentified: boolean,
         uuid: string,
         distinctIds?: string[],
-        version = 0
+        distinctIdVersions?: number[]
     ): Promise<InternalPerson> {
         distinctIds ||= []
 
+        if (distinctIdVersions === undefined) {
+            // Default to version 0 for all Distinct IDs
+            distinctIdVersions = new Array(distinctIds.length).fill(0)
+        } else {
+            // This should never happen and is only here to catch code during development/tests
+            if (distinctIds.length != distinctIdVersions.length) {
+                throw new Error('createPerson: distinctIds and distinctIdVersions lengths must match')
+            }
```

Comment:
It looks like you already [called this out over here](https://github.com/PostHog/posthog/pull/23164/files#r1649474664) but it seems like maybe the signature for this should be `{distinctId: string, version?: number}[]` or similar instead to avoid this invariant check?